templates/bond_nmconnection.j2

```diff
@@ -103,6 +103,9 @@ routing-rule{{ loop.index }}=priority {{ 32765 - loop.index0 }} {{ rule }}
 {% if item.dnsnameservers is defined %}
 dns={{ item.dnsnameservers | join(',') }}
 {% endif %}
+{% if item.dnssearch is defined %}
+dns-search={{ item.dnssearch | join(';') }};
```

Comment:
I don't think this is expecting a list, given that this is how it is used in other templates:

```
{% if item.dnssearch is defined %}
DOMAIN={{ item.dnssearch }}
{% endif %}
```

README gives this example:
```
dnssearch: example.com
```

The network-scripts and Debian interfaces formats use spaces, so we might be expecting a whitespace-separated list. You could split on whitespace then join on semicolons?

Reply:
That different templates use different separators is a reason to use lists and not strings.

In this case I copied the same implementation pattern as `item.dnsnameservers` for the previous variable.

I think a better implementation would be to handle both strings and lists, that way the list handler acts as a clue for what the string ought to be.

Reply:
That works for me.